nav2_lifecycle_manager/test/test_lifecycle_manager.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 Shivang Patel
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+#include <chrono>
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_util/node_thread.hpp"
+#include "nav2_lifecycle_manager/lifecycle_manager_client.hpp"
+#include "lifecycle_node_test.hpp"
+
+class LifecycleClientTestFixture
+{
+public:
+  LifecycleClientTestFixture()
+  {
+    lf_node_ = std::make_shared<LifecycleNodeTest>();
+    lf_thread_ = std::make_unique<nav2_util::NodeThread>(lf_node_->get_node_base_interface());
+  }
+
+private:
+  std::shared_ptr<LifecycleNodeTest> lf_node_;
+  std::unique_ptr<nav2_util::NodeThread> lf_thread_;
+};
+
+TEST(LifecycleClientTest, TestingTest)
+{
+  EXPECT_TRUE(true);
+}
+
+TEST(LifecycleClientTest, BasicTest)
+{
+  LifecycleClientTestFixture fix;
+  nav2_lifecycle_manager::LifecycleManagerClient client("lifecycle_manager_test");
+  EXPECT_TRUE(client.startup());
+  EXPECT_EQ(
+    nav2_lifecycle_manager::SystemStatus::TIMEOUT,
+    client.is_active(std::chrono::nanoseconds(1000)));
+  EXPECT_EQ(
+    nav2_lifecycle_manager::SystemStatus::ACTIVE,
+    client.is_active(std::chrono::nanoseconds(1000000000)));
+  EXPECT_TRUE(client.pause());
+  EXPECT_EQ(
+    nav2_lifecycle_manager::SystemStatus::INACTIVE,
+    client.is_active(std::chrono::nanoseconds(1000000000)));
+  EXPECT_TRUE(client.resume());
```

Comment:
So these are good tests, but you should check in the fixture if the state actually changed. You could do this by having an enum for the current state the the `on_XYZ` calls and then after calling `client.XYZ()` you `EXPECT_EQ` for getting that state. This ensures that not only the call worked, but it completed the task

Reply:
I mean it will only return true if the transition of states will be successful not the call (but the result of the call). I think the result of this ensures that check on transition.

Reply:
Oh OK, that works then